vars/handlePipelineStepErrors.groovy

```diff
@@ -81,12 +81,28 @@ void call(Map parameters = [:], body) {
         if (config.failOnError || config.stepName in config.mandatorySteps) {
             throw ex
         }
+
         if (config.stepParameters?.script) {
             config.stepParameters?.script.currentBuild.result = 'UNSTABLE'
         } else {
             currentBuild.result = 'UNSTABLE'
         }
 
+        echo "[${STEP_NAME}] Error in step ${config.stepName} - Build result set to 'UNSTABLE'"
+
+        // ONLY if commonPipelineEnvironment is shared:
```

Comment:
Not handing over the script parameters sets the build to unstable. Hence we can IMO rely on a shared script instance. See `src/com/sap/piper/Prerequisites.checkScript` which is called in all the steps. 

Would be easier to read like

```
List unstableSteps = ./.
if(! unstableSteps) {
    unstableSteps = []
}

unstableSteps.add(config.stepName)
parameters.stepParameters.script.commonPipelineEnvironment.setValue('unstableSteps', unstableSteps)
```

This is less noisy. Only what really needs to be done when no unstableSteps list is present is done in the corresponding code block. There is no need for having two mutually exclusive setter calls for `unstableSteps`.

Reply:
In this case there is no `checkScript` since the step only gets the script indirectly.

Custom script might use `handlePipelineStepErrors` without passing the `script`.

Anyway, I agree we could make it simpler and just set it on `cpe`.
This will in the end have the same effect.